src/components/learner-credit-management/tests/BudgetDetailPage.test.jsx

```diff
@@ -2301,4 +2307,106 @@ describe('<BudgetDetailPage />', () => {
     expect(screen.getByText('• Enroll via Integrated Learning Platform')).toBeInTheDocument();
     expect(screen.getByText('Manage edX in your integrated learning platform')).toBeInTheDocument();
   });
+  it.each([
+    {
+      modifiedDayOffset: 5,
+    },
+    {
+      modifiedDayOffset: 15,
+    },
+  ])('displays upcoming expiring allocation', async ({ modifiedDayOffset }) => {
```

Comment:
nit: consider including `(%s)` so the test name in terminal output includes the different `modifiedDayOffset` conditions.